### PR TITLE
Label support #9

### PR DESCRIPTION
**Pull request by kopachevsky**
Fixes terraform-google-modules/terraform-google-pubsub#9 

- Added topic_labels (map(string) terraform variable for labels.
- Updated integration test and fixtures to check label assigment.
- Switched to use example code for integration test.

### CODE REVIEW

**Comment by kopachevsky** [comment]
```
Command: `gcloud --project='gl-akopachevskyy-pubsub' pubsub topics describe tf-proj-topic`
     ✔  should match /name: projects\/gl-akopachevskyy-pubsub\/topics\/tf-proj-topic/
     ✔  should match /labels:\n  bar_label: bar_value\n  foo_label: foo_value\nname/
     ✔  exit_status should be zero
  Command: `gcloud --project='gl-akopachevskyy-pubsub' pubsub subscriptions describe pull --format=json`
     ✔  should include {:name => "projects/gl-akopachevskyy-pubsub/subscriptions/pull"}
     ✔  should include {:topic => "projects/gl-akopachevskyy-pubsub/topics/tf-proj-topic"}
     ✔  should include {:ackDeadlineSeconds => 10}
     ✔  exit_status should be zero
  Command: `gcloud --project='gl-akopachevskyy-pubsub' pubsub subscriptions describe push --format=json`
     ✔  should include {:name => "projects/gl-akopachevskyy-pubsub/subscriptions/push"}
     ✔  should include {:topic => "projects/gl-akopachevskyy-pubsub/topics/tf-proj-topic"}
     ✔  should include {:pushConfig => {:pushEndpoint => "https://gl-akopachevskyy-pubsub.appspot.com/"}}
     ✔  should include {:ackDeadlineSeconds => 20}
     ✔  exit_status should be zero
```

File: test/fixtures/outputs.tf [hunk 1]
```diff
@@ -19,5 +19,9 @@ output "project_id" {
 }
 
 output "topic_name" {
-  value = var.topic_name
+  value = module.example.topic_name
```
Comment (ingwarr) [inline test/fixtures/outputs.tf]: Add descriptions please

File: test/fixtures/outputs.tf [hunk 2]
```diff
@@ -19,5 +19,9 @@ output "project_id" {
...
+}
+
+output "topic_labels" {
```
Comment (ingwarr) [inline test/fixtures/outputs.tf]: Same

File: examples/simple/main.tf [hunk 3]
```diff
@@ -20,14 +20,15 @@ provider "google" {
...
+  source       = "../../"
+  project_id   = var.project_id
+  topic        = var.topic_name
+  topic_labels = var.topic_labels
 
   pull_subscriptions = [
     {
       name                 = "pull"
-      ack_deadline_seconds = 20
+      ack_deadline_seconds = 10
```
Comment (nick4fake) [inline examples/simple/main.tf]: why?

Comment (kopachevsky) [reply examples/simple/main.tf]: before my change integration test did not use examples, I did connect examples to test and adopt it to satisfy test asserts (see controls.rb)

File: Makefile [hunk 4]
```diff
@@ -15,8 +15,14 @@
...
 
+# Docker build config variables
+CREDENTIALS_PATH ?= /cft/workdir/credentials.json
+DOCKER_ORG := gcr.io/cloud-foundation-cicd
+DOCKER_TAG_BASE_KITCHEN_TERRAFORM ?= 2.3.0
+DOCKER_REPO_BASE_KITCHEN_TERRAFORM := ${DOCKER_ORG}/cft/kitchen-terraform:${DOCKER_TAG_BASE_KITCHEN_TERRAFORM}
+
 # All is the first target in the file so it will get picked up when you just run 'make' on its own
-all: check_shell check_python check_golang check_terraform check_docker check_base_files check_trailing_whitespace generate_docs
+all: check_shell check_python check_golang check_terraform check_base_files check_trailing_whitespace generate_docs
```
Comment (ivankorn) [inline Makefile]: Target `all` seem to be altered to include only `doc generator` in the latest versions of [project factory](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/Makefile#L29-L33) and [module template](https://github.com/terraform-google-modules/terraform-google-module-template/blob/master/terraform-google-%7B%7Bcookiecutter.module_name%7D%7D/Makefile#L29-L33) instead the checks migrated into `check` target.

Do we want to log a separate issue to back-port that change to all the modules or we do this in scope of the migration? 

cc: @aaron-lane @nick4fake @ingwarr @paulpalamarchuk

File: examples/simple/variables.tf [hunk 5]
```diff
@@ -23,3 +23,9 @@ variable "topic_name" {
   type        = string
   description = "The name for the Pub/Sub topic"
 }
+
+variable "topic_labels" {
+  type        = map(string)
+  description = "A map of labels to assign to the Pub/Sub topic"
+  default     = {}
+}
```
Comment (ivankorn) [inline examples/simple/variables.tf]: new line is missing at the end of file

File: variables.tf [hunk 6]
```diff
@@ -34,5 +40,4 @@ variable "pull_subscriptions" {
   type        = list(map(string))
   description = "The list of the pull subscriptions"
   default     = []
-}
-
+}
```
Comment (ivankorn) [inline variables.tf]: new line is missing at the end of file

File: Makefile (hunk 4 above)
Comment (aaron-lane) [reply Makefile]: This is made obsolete by the new development image and build process.

File: main.tf [hunk 7]
```diff
@@ -19,8 +19,9 @@ locals {
 }
 
 resource "google_pubsub_topic" "topic" {
-  name    = var.topic
   project = var.project_id
+  name    = var.topic
+  labels  = var.topic_labels
```
Comment (aaron-lane) [inline main.tf]: Has this been tested for backward compatibility? We need to be sure that adding labels to an existing resource isn't a destructive operation.

Comment (paulpalamarchuk) [reply main.tf]: I've tested backward compatibility. Adding labels case update in-place, without any destruction of previously-created resources: 
image

**Pull request merged**